src/Text/Pandoc/Shared.hs

```diff
@@ -1028,6 +1031,19 @@ collapseFilePath = Posix.joinPath . reverse . foldl go [] . splitDirectories
     isSingleton _ = Nothing
     checkPathSeperator = fmap isPathSeparator . isSingleton
 
+--
+-- File selection from the archive
+--
+filteredFilesFromArchive :: Archive -> (FilePath -> Bool) -> [(FilePath, BL.ByteString)]
+filteredFilesFromArchive zf f =
+  mapMaybe (fileAndBinary zf) (filter f (filesInArchive zf))
+  where
+    fileAndBinary :: Archive -> FilePath -> Maybe (FilePath, BL.ByteString)
```

Comment:
I personally would prefer to have `fileAndBinary` not inlined in order to be able to test it. Since it's Haskell and type checking already does 80% of the job verifying the program will work, decide for yourself.